generator/sbpg/targets/javascript.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# Copyright (C) 2015 Swift Navigation Inc.
+# Contact: Joshua Gross <josh@swiftnav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Generator for JavaScript target.
+
+This module consumes the YAML spec and generates some message classes
+files.
+
+"""
+
+from sbpg.targets.templating import JENV, ACRONYMS
+import copy
+
+from construct import *
+
+TEMPLATE_NAME = "sbp_js.js.j2"
+
+CONSTRUCT_CODE = {
+  'u8': 'uint8',
+  'u16': 'uint16',
+  'u32': 'uint32',
+  'u64': 'uint64',
+  's8': 'int8',
+  's16': 'int16',
+  's32': 'int32',
+  's64': 'int64',
+  'float': 'floatle',
+  'double': 'doublele',
+}
+
+WRITE_BUFFER_CODE = {
+  'u8': 'writeUInt8',
+  'u16': 'writeUInt16LE',
+  'u32': 'writeUInt32LE',
+  'u64': 'writeUIntLE', # only supports up to 48 bits of accuracy
+  's8': 'writeInt8',
+  's16': 'writeInt16LE',
+  's32': 'writeInt32LE',
+  's64': 'writeIntLE', # only supports up to 48 bits of accuracy
+  'float': 'writeFloatLE',
+  'double': 'writeDoubleLE',
+}
+
+BUFFER_SIZE_MAP = {
+  'u8': 1,
+  'u16': 2,
+  'u32': 4,
+  'u64': 8,
+  's8': 1,
+  's16': 2,
+  's32': 4,
+  's64': 8,
+  'float': 4,
+  'double': 8
+}
+
+PYDOC_CODE = {
```

Comment:
Done. I changed to the actual JavaScript type but with more detail, which I think folks will find useful. Example: `u16` now maps to `number (unsigned 16-bit int, 2 bytes)`